command-injection/row_010_YousefED_typescript-json-schema_typescript-json-schema.ts

```
@@ -0,0 +1,1879 @@
+                } else {
+                    this.getClassDefinition(typ, definition);
+                }
+            }
+        }
+
+        if (this.recursiveTypeRef.get(fullTypeName) === definition) {
+            this.recursiveTypeRef.delete(fullTypeName);
+            // If the type was recursive (there is reffedDefinitions) - lets replace it to reference
+            if (this.reffedDefinitions[fullTypeName]) {
+                const annotations = Object.entries(returnedDefinition).reduce<Record<string, unknown>>((acc, [key, value]) => {
+                    if (annotationKeywords[key as keyof typeof annotationKeywords] && typeof value !== undefined) {
```

Biome (2.1.2)

[error] Invalid typeof comparison.

Compare with one of the following string literals:


(lint/correctness/useValidTypeof)